File: Apis/Domain/Entities/UserSubscription.cs
﻿using System;
using System.Collections.Generic;

namespace Domain.Entities;

public class UserSubscription : BaseEntity
{

    public DateTime? StartDate { get; set; }

    public int? SubscriptionId { get; set; }

    public int? UserId { get; set; }

    public virtual Subscription? Subscription { get; set; }

    public virtual User? User { get; set; }
}

File: EXE201_Heureux/src/main/java/com/example/exe201_heureux/entity/UserSubscription.java
package com.example.exe201_heureux.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "User_Subscription")
public class UserSubscription {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "subscription_id")
    private Subscription subscription;

    @Column(name = "start_date")
    private LocalDate startDate;

}